SpatialGDK/Documentation/content/non-unreal-layers.md

```diff
@@ -0,0 +1,223 @@
+            const FRotator Rotation = FRotator::ZeroRotator;
+            World->SpawnActor(CubeClass, &Location, &Rotation);
+        }
+
+        Cast<USpatialNetDriver>(World->GetNetDriver())->Connection->SendCommandResponse(Op.request_id, &Response);
+    }
+}
+```
+
+### Edit the snapshot
+You could add a new entity with the given component in your Unreal project code in the following way:
+
+```
+UCLASS()
+class SPATIALGDK_API USessionEntitySnapshotGeneration : USnapshotGenerationTemplate
+{
+    GENERATED_BODY()
+
+public:
+    bool WriteToSnapshotOutput(Worker_SnapshotOutputStream* OutputStream, Worker_Entity& NextEntityId) override 
+    {
+        Worker_Entity SomeEntity;
+        SomeEntity.entity_id = NextEntityId;
+        TArray<Worker_ComponentData> Components;
+        const WorkerAttributeSet ExternalWorkerAttributeSet{ TArray<FString>{TEXT("some_external_worker_type")} };
+        const WorkerRequirementSet ExternalWorkerPermission{ ExternalWorkerAttributeSet};
+        const WorkerRequirementSet AnyWorkerPermission{ {SpatialConstants::UnrealClientAttributeSet, SpatialConstants::UnrealServerAttributeSet, ExternalWorkerAttributeSet} };
+        WriteAclMap ComponentWriteAcl;
+        ComponentWriteAcl.Add(SpatialConstants::POSITION_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+        ComponentWriteAcl.Add(SpatialConstants::METADATA_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+        ComponentWriteAcl.Add(SpatialConstants::PERSISTENCE_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+        ComponentWriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, SpatialConstants::UnrealServerPermission);
+        ComponentWriteAcl.Add(1337, ExternalWorkerPermission);
+
+        // Serialize SomeComponent data
+        Worker_ComponentData SomeComponentComponentData{};
+        FromUnrealComponentData.component_id = 1337;
+        FromUnrealComponentData.schema_type = Schema_CreateComponentData(1337);
+        Schema_Object* FromUnrealComponentDataObject = Schema_GetComponentDataFields(FromUnrealComponentData.schema_type);
+        Schema_AddInt32(FromUnrealComponentDataObject, 1, 1); // set some_counter to 1 initially
+
+        Components.Add(improbable::Position(improbable::Origin).CreatePositionData());
+        Components.Add(improbable::Metadata(TEXT("SessionManager")).CreateMetadataData());
+        Components.Add(improbable::Persistence().CreatePersistenceData());
+        Components.Add(improbable::EntityAcl(AnyWorkerPermission, ComponentWriteAcl).CreateEntityAclData());
+        Components.Add(SessionComponentData);
+        Components.Add(FromUnrealComponentData);
+
+        SomeEntity.component_count = Components.Num();
+        SomeEntity.components = Components.GetData();
+
+        bool bSuccess = Worker_SnapshotOutputStream_WriteEntity(OutputStream, &SomeEntity) != 0;
+        if (bSuccess) 
+        {
+            NextEntityId++;
+        }
+        return bSuccess;
+    }
+}
+```
```

Comment:
Please can you add to the bottom of this page:
```
<br/>
------------
2019-03-15 Page added with full editorial review

```

This is part of the Crown 1.0 clear docs and good packaging approach. Ta!